tutorial.md

```diff
@@ -0,0 +1,131 @@
+<!---
+Note: This tutorial is meant for Google Cloud Shell, and can be opened by going to
+https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/magic-modules&tutorial=tutorial.md
+--->
+# Magic Modules Tutorial
+
+## Intro
+
+This tutorial will walk you through the components that make up Magic Modules.
+
+## api.yaml
+
+Each product's api definition is stored in the magic-modules repo.
+
+Let's open
+<walkthrough-editor-open-file filePath="magic-modules/products/compute/api.yaml"
+                              text="products/compute/api.yaml">
+</walkthrough-editor-open-file>.
+
+### Product metadata
+
+The
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="!ruby/object:Api::Product"
+                                 text="top section">
+</walkthrough-editor-select-regex>
+provides metadata about the API, such as name, scopes, and versions.
+
+### Resources
+
+Each `api.yaml` file contains a list of resources.
+Let's
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="!ruby/object:Api::Resource"
+                                 text="look at">
+</walkthrough-editor-select-regex>
+the first one.
+
+This section contains data about the resource, such as its name, description, and URLs.
+
+### Parameters + Properties
+
+Each resource contains a list of
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
```

Comment:
We might want to use one of the less old products for the tutorial. Compute has lots of stuff which doesn't initially much sense (like properties listed as parameter because of overloading) or injecting other yaml files and the file is huge. Maybe redis?